### PR TITLE
Add encoding support

### PR DESCRIPTION
**Pull request by hansharhoff**
This pull request adds support for an encoding argument to test_dir (and the functions called by this). This is related to an [issue reported in devtools](https://github.com/hadley/devtools/issues/1306). A separate pull request is incoming for devtools.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Until this is merged, `devtools` or `testthat` (choose as like) is broken: it does not handle the `filter` argument in `devtools::test`, because it is passed down to `grepl`, which errors on it. 

Can you please consider reviewing it?

File: R/test-files.R [hunk 1]
```diff
@@ -27,16 +27,16 @@ test_env <- function() {
...
     source_test_helpers(path, env)
   }
   paths <- find_test_scripts(path, filter, ...)
 
-  test_files(paths, reporter = reporter, env = env, ...)
+  test_files(paths, reporter = reporter, env = env, encoding = encoding, ...)
 }
 
-test_files <- function(paths, reporter = "summary",
+test_files <- function(paths, reporter = "summary", encoding = "unknown",
```
Comment (krlmlr) [inline R/test-files.R]: How are the dots `...` used in this function?

**Comment by krlmlr** [comment]
@gaborcsardi: Did you mean the `encoding` argument?

File: R/test-files.R [hunk 2]
```diff
@@ -98,11 +99,14 @@ find_test_scripts <- function(path, filter = NULL, invert = FALSE, ...) {
 #' @param reporter reporter to use
 #' @param env environment in which to execute the tests
 #' @param load_helpers Source helper files before running the tests?
+#' @param encoding the encoding of the files in test directory. Default is
```
Comment (krlmlr) [inline R/test-files.R]: Perhaps shorter: `File encoding, default: "unknown"`

File: NEWS.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # testthat 1.0.2.9000
+* New argument`encoding` in `test_file()` and `source_file()` (@hansharhoff, devtools#1306)
```
Comment (hansharhoff) [inline NEWS.md]: Yes, 
is this better?

    (@hansharhoff, hadley/devtools#1306)

File: R/test-files.R (hunk 1 above)
Comment (hansharhoff) [inline R/test-files.R]: I am not sure I completely understand. I wanted to keep it as close as possible to the parent.

**Comment by hansharhoff** [comment]
I believe I addressed the points of @krlmlr with this commit. If you agree I will ping hadley.

File: R/test-files.R (hunk 1 above)
Comment (krlmlr) [inline R/test-files.R]: The dots `...` currently are not used by `test_files()`. They weren't before, but as we're touching the code now, it's better to clean this up.

**Comment by hansharhoff** [comment]
@krlmlr , I have removed the dots and simplified the arguments. I take it those were the open comments.

File: R/test-files.R [hunk 4]
```diff
@@ -98,11 +99,14 @@ find_test_scripts <- function(path, filter = NULL, invert = FALSE, ...) {
 #' @param reporter reporter to use
 #' @param env environment in which to execute the tests
 #' @param load_helpers Source helper files before running the tests?
+#' @param encoding File encoding, default is "unknown"
```
Comment (krlmlr) [inline R/test-files.R]: Can you use `@inheritParams source_file` here?

**Comment by krlmlr** [comment]
@hadley: PTAL.

File: R/source.R [hunk 5]
```diff
@@ -8,12 +8,13 @@
...
 #' @param chdir Change working directory to \code{dirname(path)}?
+#' @param encoding File encoding, default: "unknown"
 #' @export
-source_file <- function(path, env = test_env(), chdir = TRUE) {
+source_file <- function(path, env = test_env(), chdir = TRUE, encoding = "unknown") {
   stopifnot(file.exists(path))
   stopifnot(is.environment(env))
 
-  lines <- readLines(path, warn = FALSE)
+  lines <- readLines(path, warn = FALSE, encoding = encoding)
```
Comment (hadley) [inline R/source.R]: The encoding argument does not do what you think it does...

**Pull request closed without merging**